src/main/java/budgetbuddy/model/account/Account.java

```diff
@@ -48,16 +50,43 @@ public Transaction getTransaction(Index toGet) {
         return this.transactionList.getTransaction(toGet);
     }
 
+    public long getBalance() {
+        return balance;
+    }
+
+    /**
+     * Add a transaction to the transactionList.
+     * @param toAdd
+     */
     public void addTransaction(Transaction toAdd) {
         this.transactionList.add(toAdd);
+        if (toAdd.getDirection().toString().equals("IN")) {
+            balance = balance + toAdd.getAmount().toLong();
+        } else {
+            balance = balance - toAdd.getAmount().toLong();
+        }
     }
 
+    /**
+     * Update the transaction.
+     * @param txnIndex
+     * @param editedTxn
+     */
     public void updateTransaction(Index txnIndex, Transaction editedTxn) {
         this.transactionList.setTransaction(txnIndex, editedTxn);
     }
 
+    /**
+     * Deletes a transaction from the transactionList.
+     * @param toDelete
+     */
     public void deleteTransaction(Transaction toDelete) {
         this.transactionList.remove(toDelete);
+        if (toDelete.getDirection().toString().equals("IN")) {
+            balance = balance - toDelete.getAmount().toLong();
+        } else {
+            balance = balance - toDelete.getAmount().toLong();
+        }
     }
 
     @Override
```

Comment:
You might want to change the `equals()` method to check for `balance` as well now.